dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGVisualizer.java

```diff
@@ -48,116 +49,135 @@
      * call back to this visualizer instance for sub-components.
      *
      * @param store the store to visualize
+     * @return the String representation of the store {@code store}
      */
-    void visualizeStore(S store);
+    String visualizeStore(S store);
 
     /**
      * Called by a {@code CFAbstractStore} to visualize the class name before calling the {@code
      * CFAbstractStore#internalVisualize()} method.
      *
      * @param classCanonicalName the canonical name of the class
+     * @return the String representation of the header to use for a store with the given canonical
+     *     name
      */
-    void visualizeStoreHeader(String classCanonicalName);
+    String visualizeStoreHeader(String classCanonicalName);
 
     /**
-     * Called by {@code CFAbstractStore#internalVisualize()} to visualize a local variable.
+     * Called by {@code CFAbstractStore#internalVisualize()} to visualize a local variable in the
+     * given store.
```

Comment:
You should use "the given store" when a store is a formal parameter.  That does not seem to be the case here.  What store are you referring to?
The same comment applies elsewhere.

Reply:
Thanks for your comments. Here is my thinking. This method is one of the methods called by `internalVisualize()` to generate a part of the visualisation of the store which is the parameter of `visualizeStore(Store store)`. "the given store" does not appear as a parameter at here, but this method is used to support the visualisation of "the given store".

For example, if we call
```java
   viz.visualizeStore(store)
```
Then this method is called to visualize "the given store":
```java
    @Override
    public String visualize(CFGVisualizer<?, S, ?> viz) {
        /* This cast is guaranteed to be safe, as long as the CFGVisualizer is created by
         * CFGVisualizer<Value, Store, TransferFunction> createCFGVisualizer() of GenericAnnotatedTypeFactory */
        @SuppressWarnings("unchecked")
        CFGVisualizer<V, S, ?> castedViz = (CFGVisualizer<V, S, ?>) viz;
        StringBuilder sbVisualize = new StringBuilder();
        sbVisualize.append(castedViz.visualizeStoreHeader(this.getClass().getCanonicalName()));
        sbVisualize.append(internalVisualize(castedViz));
        sbVisualize.append(castedViz.visualizeStoreFooter());
        return sbVisualize.toString();
    }
```
We can find that `internalVisualize(castedViz)` is called at here.
And here is what `internalVisualize(castedViz)` does:
```java
    /**
     * Adds a representation of the internal information of this Store to visualizer {@code viz}.
     *
     * @return a String of the internal information of this {@link Store}
     */
    protected String internalVisualize(CFGVisualizer<V, S, ?> viz) {
        StringBuilder res = new StringBuilder();
        for (Entry<FlowExpressions.LocalVariable, V> entry : localVariableValues.entrySet()) {
            res.append(viz.visualizeStoreLocalVar(entry.getKey(), entry.getValue()));
        }
        if (thisValue != null) {
            res.append(viz.visualizeStoreThisVal(thisValue));
        }
        for (Entry<FlowExpressions.FieldAccess, V> entry : fieldValues.entrySet()) {
            res.append(viz.visualizeStoreFieldVals(entry.getKey(), entry.getValue()));
        }
        for (Entry<FlowExpressions.ArrayAccess, V> entry : arrayValues.entrySet()) {
            res.append(viz.visualizeStoreArrayVal(entry.getKey(), entry.getValue()));
        }
        for (Entry<MethodCall, V> entry : methodValues.entrySet()) {
            res.append(viz.visualizeStoreMethodVals(entry.getKey(), entry.getValue()));
        }
        for (Entry<FlowExpressions.ClassName, V> entry : classValues.entrySet()) {
            res.append(viz.visualizeStoreClassVals(entry.getKey(), entry.getValue()));
        }
        return res.toString();
    }
```
This method calls several `visualizeStoreXXX` methods and combine the return value of them as the String representation of "the given store".

Reply:
`visualizeStoreHeader` and `visualizeStoreFooter` are called in the same way. They return the String representation of the header and the footer for "the given store" which is the parameter in `visualizeStore(Store store)`.

Reply:
What about I remove the "store" part, will it be helpful?

Something like:
```java
  /**
     * Called by {@code CFAbstractStore#internalVisualize()} to visualize a local variable.
     *
     * @param localVar the local variable
     * @param value the value of the local variable
     * @return the String representation of the local variable
     */
```